src/backend/metal/src/command.rs

```diff
@@ -1335,47 +1335,18 @@ impl com::RawCommandBuffer<Backend> for CommandBuffer {
         T::Item: Borrow<SubresourceRange>,
     {
         let mut inner = self.inner.borrow_mut();
+        let clear_color = image.shader_channel.interpret(color);
 
-        let clear_color = unsafe {
-            match image.shader_channel {
-                Channel::Float => metal::MTLClearColor::new(
-                    color.float32[0] as _,
-                    color.float32[1] as _,
-                    color.float32[2] as _,
-                    color.float32[3] as _,
-                ),
-                Channel::Int => metal::MTLClearColor::new(
-                    color.int32[0] as _,
-                    color.int32[1] as _,
-                    color.int32[2] as _,
-                    color.int32[3] as _,
-                ),
-                Channel::Uint => metal::MTLClearColor::new(
-                    color.uint32[0] as _,
-                    color.uint32[1] as _,
-                    color.uint32[2] as _,
-                    color.uint32[3] as _,
-                ),
-            }
-        };
+        //TODO: this is possible to optimize by bulk-cleaning all layers with:
+        // `descriptor.set_render_target_array_length(sub.layers.end as _);`
+        // this only works when `sub.layers.start == 0`, so one could create a temporary
+        // texture view to work around that.
 
         for subresource_range in subresource_ranges {
             let sub = subresource_range.borrow();
-            let end_level = if sub.levels.end == !0 {
-                image.raw.mipmap_level_count() as _
-            } else {
-                sub.levels.end
-            };
-            let end_layer = if sub.layers.end == !0 {
-                image.raw.array_length() as _
```

Comment:
Why don't we have this case anymore?

Reply:
it's handled in `portability` now, no special values here any more :tada: